services/app-api/handlers/reports/create.ts

```diff
@@ -101,91 +109,78 @@ export const createReport = handler(
 
-    let workPlanMetadata: ReportMetadata | undefined = undefined;
-
-    // Get Associated Work Plan if creating a SAR Submission
-    if (reportType === ReportType.SAR) {
-      const sarSubmissions = await fetchReportsByState(event, _context);
-
-      const workPlanEvent = event;
-      workPlanEvent.pathParameters = {
-        ...workPlanEvent.pathParameters,
-        reportType: ReportType.WP,
-      };
-      const workPlanSubmissions = await fetchReportsByState(event, _context);
-      if (!workPlanSubmissions.body) {
-        return {
-          status: StatusCodes.NOT_FOUND,
-          body: error.NO_WORKPLANS_FOUND,
-        };
-      }
-      // if current report exists, parse for archived status
-      if (workPlanSubmissions.body) {
-        const currentSarSubmissions = JSON.parse(sarSubmissions.body);
-        const currentWorkPlans = JSON.parse(workPlanSubmissions.body);
-
-        if (currentWorkPlans.length !== currentSarSubmissions?.length) {
-          let lastFoundSubmission: ReportMetadataShape | undefined = undefined;
-          currentWorkPlans.forEach((submission: ReportMetadataShape) => {
-            if (
-              submission.status === ReportStatus.NOT_STARTED ||
-              submission.status === ReportStatus.IN_PROGRESS
-            ) {
-              if (
-                lastFoundSubmission &&
-                submission.createdAt > lastFoundSubmission?.createdAt
-              )
-                lastFoundSubmission = submission;
-              else if (!lastFoundSubmission) {
-                lastFoundSubmission = submission;
-              }
-            }
-          });
-
-          if (lastFoundSubmission) {
-            const fetchWPReportEvent = event;
-            fetchWPReportEvent.pathParameters = {
-              reportType: ReportType.WP,
-              state: state,
-              id: lastFoundSubmission["id"],
-            };
-            const workPlan = await fetchReport(fetchWPReportEvent, _context);
-            const workPlanParsed = JSON.parse(workPlan.body);
-            workPlanMetadata = workPlanParsed;
-            const workPlanFieldData = workPlanParsed?.fieldData;
-            validatedFieldData = {
-              ...validatedFieldData,
-              workPlanData: workPlanFieldData,
-            };
-          }
-        }
-      }
-    }
-
     // Return INVALID_DATA error if field data is not valid.
     if (!validatedFieldData || Object.keys(validatedFieldData).length === 0) {
       return {
         status: StatusCodes.SERVER_ERROR,
         body: error.INVALID_DATA,
       };
     }
+    // End Section - Check the payload that was sent with the request and validate it
+
+    /*
+     * Begin Section - If creating a SAR Submission, find the last Work Plan created that hasn't been used
+     * to create a different SAR and attach all of its fieldData to the SAR Submissions FieldData
+     */
+    const {
+      workPlanMetadata,
+      workPlanFieldData,
+    }:
+      | {
```

Comment:
is it possible to simplify this to: 

```
const {
      workPlanMetadata,
      workPlanFieldData,
    }:
      {
          workPlanMetadata?: ReportMetadataShape;
          workPlanFieldData?: AnyObject;
        } =
      reportType === ReportType.SAR
        ? await getLastCreatedWorkPlan(event, _context, state)
        : { workPlanMetadata: undefined, workPlanFieldData: undefined };
```

or does it get wonky and think `?` means null instead of optionally undefined?

Reply:
Your so right, thank you. Fixed!